src/refract/Element.cc

```diff
@@ -129,4 +131,276 @@ namespace refract
         }
     }
 
+    namespace {
+        struct TypeChecker {
```

Comment:
I understood what this is about but I think it's better if we have a comment in the code explaining this.